ice/src/main/java/com/altinity/ice/cli/Main.java

```diff
@@ -426,6 +428,42 @@ void deleteNamespace(
     }
   }
 
+  @CommandLine.Command(name = "delete", description = "Delete Partition(s).")
+  void deletePartition(
+      @CommandLine.Parameters(
+              arity = "1",
+              paramLabel = "<name>",
+              description = "Table name (e.g. ns1.table1)")
+          String name,
+      @CommandLine.Option(
+              names = {"--partition"},
+              description =
+                  "JSON array of partition filters: [{\"name\": \"vendorId\", \"values\": [5, 6]}]. "
+                      + "For timestamp columns, use ISO Datetime format YYYY-MM-ddTHH:mm:ss")
+          String partitionJson,
+      @CommandLine.Option(
+              names = "--dry-run",
+              description = "Log files that would be deleted without actually deleting them")
+          boolean dryRun)
```

Comment:
might be worth it to have it true be default considering the nature of the operation